packages/services/api/src/modules/audit-logs/providers/audit-logs-manager.ts

```diff
@@ -0,0 +1,326 @@
+import { stringify } from 'csv-stringify';
+import { GraphQLError } from 'graphql';
+import { Inject, Injectable, Scope } from 'graphql-modules';
+import { z } from 'zod';
+import { captureException } from '@sentry/node';
+import { Session } from '../../auth/lib/authz';
+import { AuthManager } from '../../auth/providers/auth-manager';
+import { ClickHouse, sql } from '../../operations/providers/clickhouse-client';
+import { SqlValue } from '../../operations/providers/sql';
+import { Emails, mjml } from '../../shared/providers/emails';
+import { Logger } from '../../shared/providers/logger';
+import { S3_CONFIG, type S3Config } from '../../shared/providers/s3-config';
+import { auditLogSchema } from './audit-logs-types';
+
+const auditLogEventTypes = auditLogSchema.options.map(option => option.shape.eventType.value);
+
+export const AuditLogClickhouseObjectModel = z.object({
+  id: z.string(),
+  timestamp: z.string(),
+  organization_id: z.string(),
+  event_action: z.enum(auditLogEventTypes as [string, ...string[]]),
+  user_id: z.string(),
+  user_email: z.string(),
+  metadata: z.string().transform(x => JSON.parse(x)),
+});
+
+export type AuditLogType = z.infer<typeof AuditLogClickhouseObjectModel>;
+
+const AuditLogClickhouseArrayModel = z.array(AuditLogClickhouseObjectModel);
+
+@Injectable({
+  scope: Scope.Operation,
+})
+/**
+ * Responsible for accessing audit logs.
+ */
+export class AuditLogManager {
+  private logger: Logger;
+
+  constructor(
+    logger: Logger,
+    private clickHouse: ClickHouse,
+    @Inject(S3_CONFIG) private s3Config: S3Config,
+    private emailProvider: Emails,
+    private session: Session,
+    private auth: AuthManager,
+  ) {
+    this.logger = logger.child({ source: 'AuditLogManager' });
+  }
+
+  async getPaginatedAuditLogs(
+    organizationSlug: string,
+    pagination?: { first: number; after: number },
+    filter?: { startDate?: Date; endDate?: Date },
+  ): Promise<{ data: AuditLogType[] }> {
+    this.logger.info(
+      'Getting audit logs (organizationId=%s, filter=%o, pagination=%o)',
+      organizationSlug,
+      filter,
+      pagination,
+    );
+
+    const isOwner = await this.auth.isOwnerOfOrganization({
+      organization: organizationSlug,
+    });
+    if (!isOwner) {
+      throw new GraphQLError('Unauthorized: You are not authorized to perform this action');
+    }
+
+    const sqlFirst = sql.raw(pagination?.first ? pagination.first.toString() : '25');
+    const sqlAfter = sql.raw(pagination?.after ? pagination.after.toString() : '0');
```

Comment:
TODO @TuvalSimha Please use cursor based pagination not offset based pagination.

Example: https://github.com/graphql-hive/console/blob/914c9cd1c0ed1d154e587a1dc3806cafd3fcc8a0/packages/services/api/src/modules/app-deployments/providers/app-deployments.ts#L707

Also for explaining the differences: https://the-guild.dev/blog/graphql-cursor-pagination-with-postgresql